bugsnag-android-core/src/main/java/com/bugsnag/android/User.kt

```diff
@@ -25,12 +26,41 @@ class User @JvmOverloads internal constructor(
     @Throws(IOException::class)
     override fun toStream(writer: JsonStream) {
         writer.beginObject()
-        writer.name("id").value(id)
-        writer.name("email").value(email)
-        writer.name("name").value(name)
+        writer.name(KEY_ID).value(id)
+        writer.name(KEY_EMAIL).value(email)
+        writer.name(KEY_NAME).value(name)
         writer.endObject()
     }
 
+    internal companion object: JsonReadable<User> {
+        private const val KEY_ID = "id"
+        private const val KEY_NAME = "name"
+        private const val KEY_EMAIL = "email"
+
+        override fun fromReader(reader: JsonReader): User? {
+            var user: User? = null
+            with(reader) {
+                beginObject()
+                var id: String? = null
+                var email: String? = null
+                var name: String? = null
+
+                while (hasNext()) {
+                    val key = nextName()
+                    val value = nextString()
+                    when (key) {
+                        KEY_ID -> id = value
+                        KEY_EMAIL -> email = value
+                        KEY_NAME -> name = value
+                    }
+                    user = User(id, email, name)
+                }
+                endObject()
```

Comment:
This constructs a new User three times - any reason it cannot be done after the end of the object and just returned?

```suggestion
                while (hasNext()) {
                    val key = nextName()
                    val value = nextString()
                    when (key) {
                        KEY_ID -> id = value
                        KEY_EMAIL -> email = value
                        KEY_NAME -> name = value
                    }
                }
                endObject()
                return User(id, email, name)
```

Reply:
Nice catch - the intention was to create the object only once.